cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -723,7 +723,11 @@ public MapAndStatus(T loadMap, K status) {
                                                     KafkaRebalance kafkaRebalance,
                                                     KafkaRebalanceAnnotation rebalanceAnnotation,
                                                     RebalanceOptions.RebalanceOptionsBuilder rebalanceOptionsBuilder) {
-        if (rebalanceAnnotation == KafkaRebalanceAnnotation.refresh) {
+
+        if (rebalanceAnnotation == KafkaRebalanceAnnotation.refresh
```

Comment:
I think it would be better to move this check out to a private method (something like `shouldRenewRebalance` or similar) that returns a boolean. So we just have `if shouldRenewRebalance(rebalanceAnnotation)`.